packages/hub/src/components/GlobalSearch/index.tsx

```diff
@@ -0,0 +1,175 @@
+    if (result.result.__typename === "BaseError") {
+      return [
+        {
+          type: "error",
+          message: result.result.message,
+        },
+      ];
+    } else if (result.result.__typename !== "QuerySearchConnection") {
+      return [
+        {
+          type: "error",
+          message: "Unknown error",
+        },
+      ];
+    }
+    return result.result.edges.map((edge) => ({
+      type: "object",
+      id: edge.node.id,
+      link: edge.node.link,
+      edge,
+      object: edge.node.object,
+    }));
+  };
+
+  // https://github.com/JedWatson/react-select/discussions/4669#discussioncomment-1994888
+  const ref = useRef<SelectInstance<SearchOption> | null>(null);
+
+  const handleKeyPress = useCallback((event: KeyboardEvent) => {
+    if (event.key === "Escape") {
+      // I assume this is fast because it will be called on each "Escape" press, even when the search is not focused.
+      ref.current?.blur();
+    }
+    if (event.metaKey && event.key === "k") {
+      event.preventDefault();
+      event.stopPropagation();
+      ref.current?.focus();
+    }
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener("keydown", handleKeyPress);
+
+    return () => {
+      document.removeEventListener("keydown", handleKeyPress);
+    };
+  }, [handleKeyPress]);
+
+  return (
+    <AsyncSelect<SearchOption>
+      ref={ref}
+      unstyled
+      loadOptions={loadOptions}
+      components={{
+        Option: SearchResult,
+        DropdownIndicator,
+      }}
+      isOptionDisabled={(option) => option.type === "error"}
+      closeMenuOnSelect
+      blurInputOnSelect
+      openMenuOnClick={false}
+      placeholder="Search..."
+      getOptionValue={(option) =>
+        option.type === "error" ? "error" : option.id
+      }
+      getOptionLabel={(option) =>
+        option.type === "error" ? "error" : option.id
+      }
+      onChange={(option) => {
+        if (option?.type === "object") {
+          router.push(option.link);
+        }
+      }}
+      controlShouldRenderValue={false}
+      classNames={{
+        // copy-pasted and simplified from SelectFormField
+        control: () =>
+          "min-w-[12em] max-w-[12em] !min-h-0 h-8 cursor-pointer bg-slate-200 hover:bg-white focus-within:bg-white transition-colors border-slate-300 border rounded-md shadow-sm focus-within:ring-indigo-400 focus-within:ring-2",
+        // disable default browser focus style
+        input: () => "[&_input:focus]:!ring-transparent",
+        placeholder: () => "text-slate-400 text-sm font-light",
+        valueContainer: () => "px-3",
+        clearIndicator: () => "text-slate-300 hover:text-slate-500 px-2",
+        loadingIndicator: () => "text-slate-300 hover:text-slate-500 px-2",
+        indicatorSeparator: () => "hidden",
+        dropdownIndicator: () => "text-slate-300 hover:text-slate-500 px-2",
+        menuPortal: () => "!z-[100]",
+        // based on Dropdown styles
+        menu: () =>
```

Comment:
Another nice-to-have - I'm curious if there's a way we could put much of this in ``packages/components``, then test it out with Storybook. I could see spending some time polishing this.

Reply:
We could make a storybook for Hub. Or move most of these styles to `packages/ui`.